### PR TITLE
Add support for bad_words Issue:#246

### PR DESCRIPTION
**Pull request by ujwal-narayan**
Added support for bad_word_ids as discussed in Issue#246 
- Modified GENSettings to allow users to provide a list of bad words/phrases
- Modified generate_text in `happy-transformer/happy_generation.py` 
- Added a test_case `test_bad_words` in `tests/test_gen.py`
- Updated documentation and examples  in `docs/1-text-generation/3-settings.md`

### CODE REVIEW

**Review by EricFillion — CHANGES_REQUESTED** [review]
Thanks for adding this feature! I really appreciate how you put in effort to write in-depth tests and to update the documentation. Here are some suggestions

File: happytransformer/happy_generation.py [hunk 1]
```diff
@@ -82,7 +84,10 @@ def generate_text(self, text: str, args: GENSettings=GENSettings()) -> Generatio
         input_ids = self.tokenizer.encode(text, return_tensors="pt")
         adjusted_min_length = args.min_length + len(input_ids[0])
         adjusted_max_length = args.max_length + len(input_ids[0])
-
+        if args.bad_words:
+            bad_words_ids = [self.tokenizer(" "+phrase.strip()).input_ids for phrase in args.bad_words]
```
[the remark this answers is not in the document]
Comment (ujwal-narayan) [reply happytransformer/happy_generation.py:88]: In the tokenizer, a word occuring at  the start of the sentence is treated differently from  a word occuring in the middle of the sentence.  So without the white space, the first word will have a different word id and thus won't actually be prohibited from generation. 
```python
>>> from transformers import GPT2TokenizerFast
>>> tokenizer = GPT2TokenizerFast.from_pretrained("gpt2")
>>> tokenizer("Hello world")['input_ids']
[15496, 995]
>>> tokenizer(" Hello world")['input_ids']
[18435, 995]
```

With the regular slow tokenization, you can have a parameter`add_prefix_space=True` at inference time to essentially do this. But with the FastTokenizer variant, you would need to initialize the tokenizer with the `add_prefix_space=True`, and thus you can't do it at inference time. 

I wasn't keen on changing the instantiation of the tokenizer as I didn't know what else would change with it, so I manually added the prefix space.

**Comment by ujwal-narayan** [comment]
Thank you so much for your feedback!  I've  updated the pull request with the requested changes :)

**Comment by EricFillion** [comment]
Thanks for adding this feature! I'll create a new release within a week.

File: happytransformer/happy_generation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (EricFillion) [reply happytransformer/happy_generation.py:88]: Great explanation! What you did sounds reasonable

**Pull request merged**